vnets.tf

```diff
@@ -58,14 +55,33 @@ resource "azurerm_virtual_network" "private" {
   tags                = local.default_tags
 }
 
-# Dedicated subnet for external access
+# Dedicated subnet for external access (such as VPN external NIC)
 resource "azurerm_subnet" "dmz" {
   name                 = "${azurerm_virtual_network.private.name}-dmz"
   resource_group_name  = azurerm_resource_group.private.name
   virtual_network_name = azurerm_virtual_network.private.name
   address_prefixes     = ["10.248.0.0/28"]
 }
 
+# Dedicated subnet for machine to machine private communications
+resource "azurerm_subnet" "data_tier" {
+  name                 = "${azurerm_virtual_network.private.name}-data-tier"
+  resource_group_name  = azurerm_resource_group.private.name
+  virtual_network_name = azurerm_virtual_network.private.name
+  address_prefixes     = ["10.248.1.0/24"]
+}
+
+# Dedicated subnet for the  "privatek8s" AKS cluster resources
+## Important: the "terraform-production" Enterprise Application used by this repo pipeline needs to be able to manage this subnet.
+## See the corresponding role assignment for this cluster added here (private repo):
+## https://github.com/jenkins-infra/terraform-states/blob/1f44cdb8c6837021b1007fef383207703b0f4d76/azure/main.tf#L49
+resource "azurerm_subnet" "privatek8s_tier" {
+  name                 = "privatek8s-tier"
+  resource_group_name  = azurerm_resource_group.private.name
+  virtual_network_name = azurerm_virtual_network.private.name
+  address_prefixes     = ["10.249.0.0/16"]
```

Comment:
Why /16?

Reply:
it was the AKS sizing (this network hosts all AKS nodes, pods, Kubernetes Service Cluster IPs so requires a LOT of IP to be avaible).

Gotta add a comment to AKS doc to confirm this AND make it easier for later.